src/truncate.jl

```diff
@@ -146,7 +146,7 @@ end
 
 ### evaluation
 
-quantile(d::Truncated, p::Real) = quantile(d.untruncated, d.lcdf + p * d.tp)
+quantile(d::Truncated, p::T) where {T<:Real} = min(T(d.upper), quantile(d.untruncated, d.lcdf + p * d.tp))
```

Comment:
There are a few problems with the suggested code:
- `d.upper` might be `nothing`
- it might not be possible to convert `d.upper` to `T` (e.g., if `T = Int` and `d.upper = 1.2`)
- it does not cover `p = 0`.

Reply:
I think we might want to use something like
```suggestion
function quantile(d::Truncated, p::Real)
    x = quantile(d.untruncated, d.lcdf + p * d.tp)
    min_x, max_x = extrema(d)
    return clamp(x, oftype(x, min_x), oftype(x, max_x))
end
```